dace/transformation/passes/dead_dataflow_elimination.py

```diff
@@ -137,16 +139,23 @@ def apply_pass(self, sdfg: SDFG, pipeline_results: Dict[str, Any]) -> Optional[D
                                     predecessor_nsdfgs[leaf.src].add(leaf.src_conn)
 
                             # Pruning connectors on tasklets sometimes needs to change their code
-                            elif (isinstance(leaf.src, nodes.Tasklet)
-                                  and leaf.src.code.language != dtypes.Language.Python):
+                            elif isinstance(leaf.src, nodes.Tasklet):
+                                ctype = infer_types.infer_out_connector_type(sdfg, state, leaf.src, leaf.src_conn)
+                                # Add definition
                                 if leaf.src.code.language == dtypes.Language.CPP:
-                                    ctype = infer_types.infer_out_connector_type(sdfg, state, leaf.src, leaf.src_conn)
                                     if ctype is None:
                                         raise NotImplementedError(
                                             f'Cannot eliminate dead connector "{leaf.src_conn}" on '
                                             'tasklet due to connector type inference failure.')
-                                    # Add definition
                                     leaf.src.code.code = f'{ctype.as_arg(leaf.src_conn)};\n' + leaf.src.code.code
+                                elif leaf.src.code.language == dtypes.Language.Python:
+                                    if ctype is not None:
+                                        ast_find = astutils.ASTFindReplace({}, trigger_names={leaf.src_conn})
+                                        try:
+                                            for code in leaf.src.code.code:
+                                                ast_find.generic_visit(code)
+                                        except astutils.NameFound:
+                                            leaf.src.code.code = [ast.parse(f'{leaf.src_conn}: dace.{ctype.to_string()}\n')] + leaf.src.code.code
```

Comment:
This will likely not work well, the type of the AST node that ast.parse results in is ast.Module.